### PR TITLE
added add_header method

### PR DESCRIPTION
**Pull request by willmcgugan**
**What this PR solves**

- Adds an `add_header` method to add an arbitrary header to a websocket request.

**How it is done**

-

**What to look out for**

-

**Screenshots (if appropriate)**

-

**Review**

- [x] Ready for review

### CODE REVIEW

File: lomond/websocket.py [hunk 1]
```diff
@@ -126,6 +128,19 @@ def __exit__(self, exc_type, exc_value, traceback):
...
         self.session.close()
 
+    def add_header(self, header, value):
+        """Add a user-defined header to the websocket request.
+
+        :param bytes header: Name of the header.
+        :param bytes value: Value of the header.
+
+        """
+        if not isinstance(header, bytes):
```
Comment (willmcgugan) [inline lomond/websocket.py]: Intentionally not doing any 'helpful' encoding of strs there. It invariably hides bugs that pop up as encoding errors in code you don't understand.

File: lomond/websocket.py [hunk 2]
```diff
@@ -287,14 +302,16 @@ def build_request(self):
             "GET {} HTTP/1.1".format(self.resource).encode('utf-8')
         ]
         version = '{}'.format(constants.WS_VERSION)
-        headers = [
+        headers = self._headers[:]
```
Comment (willmcgugan) [inline lomond/websocket.py]: This maybe simplistic as the dev might attempt to set the automatically generated headers. It will suffice for now.

**Comment by willmcgugan** [comment]
@wildfoundry/web-developers CI failing due to some configuration error, but I'd appreciate a review.

**Comment by willmcgugan** [comment]
> I just wondered - is it a concern for us that somebody might call add_header twice and we would end up with two headers for different values ?

True. Although that's actually valid for some headers that accept coma separated values.

For instance, this:

```
foo: bar
foo: egg
```

should be equivalent to:

```
foo: bar, egg
```

At some point, I might add smarter http header building, but I just need a quick addition to add a an 'Authorization' header.

File: tests/test_websocket.py [hunk 3]
```diff
@@ -107,6 +107,27 @@ def test_build_request(websocket):
     )
 
 
+def test_build_request_custom_headers(websocket):
+    with pytest.raises(TypeError):
+        websocket.add_header(1, 2)
+    with pytest.raises(TypeError):
+        websocket.add_header(b'foo', 2)
+    websocket.add_header(b'foo', b'bar')
```
Comment (meshy) [inline tests/test_websocket.py]: These seem sensible to me as three tests, one for each invalid param, and one for the expected behaviour of a valid call. Not a blocker though.

**Pull request merged**